src/primitives/sphere.jl

```diff
@@ -37,6 +37,30 @@ end
 
 Sphere(p1::Tuple, p2::Tuple, p3::Tuple) = Sphere(Point(p1), Point(p2), Point(p3))
 
+"""
+    Sphere(p1, p2, p3, p4)
+
+A 3D sphere passing through points `p1`, `p2`, `p3` and `p4`.
+"""
+function Sphere(p1::Point{3}, p2::Point{3}, p3::Point{3}, p4::Point{3})
+  vol = volume(Tetrahedron(p1, p2, p3, p4))
+  T = typeof(vol)
+  if isapprox(vol, zero(T); atol = atol(T))
+    throw(ArgumentError("The four points are coplanar."))
+  end
+  v1 = p1 - p4
+  v2 = p2 - p4
+  v3 = p3 - p4
+  u = (v3 ⋅ v3) * (v1 × v2) + (v2 ⋅ v2) * (v3 × v1) + (v1 ⋅ v1) * (v2 × v3)
```

Comment:
We can probably express this big expression in terms of volumes? What is the intuition behind this set of cross products? Perhaps there is a determinant hidden here too?

Reply:
Try to think in terms of geometric concepts. This linear algebra stuff becomes an implementation detail. If we can write code in terms of geometries it will be much easier to generalize it in the future.

Reply:
No idea.

Reply:
It is taking the planes v1xv2, v3xv1 and v2xv3 and scaling by the other vectors. If there was a minus sign somewhere we could try to map to determinant, but at a first glance it is not something straightforward.